package/MDAnalysis/core/topologyattrs.py

```diff
@@ -473,8 +473,65 @@ def _gen_initial_values(na, nr, ns):
         return np.arange(1, na + 1)
 
 
+class _AtomStringAttr(AtomAttr):
+    def __init__(self, vals, guessed=False):
+        self._guessed = guessed
+      
+        self.namedict = dict()  # maps str to nmidx
+        name_lookup = []  # maps idx to str
+        # eg namedict['O'] = 5 & name_lookup[5] = 'O'
+
+        self.nmidx = np.zeros_like(vals, dtype=int)  # the lookup for each atom
```

Comment:
I would not mind a more explicit name than `nmidx`. `name_index` perhaps? But there may be something even better I do not think of.

Reply:
Does`nmidx` need to be public or could it just be hidden behind an underscore?